seam/seam_multi_workspace.py

```diff
@@ -0,0 +1,71 @@
+from typing import Dict, Optional, Union
+from typing_extensions import Self
+
+from seam.auth import get_auth_headers_for_multi_workspace_personal_access_token
+from seam.constants import LTS_VERSION
+from seam.options import get_endpoint
+from seam.request import RequestMixin
+from seam.types import AbstractSeamMultiWorkspace
+from seam.workspaces import Workspaces
+
+
+class SeamMultiWorkspace(AbstractSeamMultiWorkspace, RequestMixin):
+    """
+    Seam class used to interact with Seam API without being scoped to any specific workspace.
+    """
+
+    lts_version: str = LTS_VERSION
+
+    def __init__(
+        self,
+        personal_access_token: str,
+        *,
+        endpoint: Optional[str] = None,
+        wait_for_action_attempt: Optional[Union[bool, Dict[str, float]]] = False,
+    ):
+        """
+        Parameters
+        ----------
+        personal_access_token : str, optional
+          Personal access token.
+        endpoint : str, optional
+          The API endpoint to which the request should be sent.
+        wait_for_action_attempt : bool or dict, optional
+          Controls whether to wait for an action attempt to complete, either as a boolean or as a dictionary specifying `timeout` and `poll_interval`. Defaults to `False`.
+        """
+
+        self.lts_version = SeamMultiWorkspace.lts_version
+        self.wait_for_action_attempt = wait_for_action_attempt
+        self._auth_headers = get_auth_headers_for_multi_workspace_personal_access_token(
+            personal_access_token
+        )
+        self._endpoint = get_endpoint(endpoint)
+
+        self._workspaces = Workspaces(seam=self)
+        self.workspaces = self.WorkspacesProxy(self._workspaces)
+
+    class WorkspacesProxy:
```

Comment:
Can this be defined outside the class (statically vs dynamically)?

Reply:
63737145ae61f319b2b72994d8a27788ca3ecf3f

Reply:
Does it make sense to define `WorkspacesProxy ` in the same file? If there are more endpoints supporting multi workspace, we'll just add new proxy classes to the file?

Reply:
This is ok for now, there is probably a way to generate this based on the supported auth methods. Might be easier to experiment with that in the js repo first.